app/models/workers_scaling/experiment_statistics.rb

```diff
@@ -0,0 +1,121 @@
+require 'workers_scaling/utils/query'
+module WorkersScaling
+  ##
+  # ExperimentStatistics class calculates various statistics about
+  # experiment run and execution.
+  class ExperimentStatistics
+
+    ##
+    # Number of running simulations in sequential worker
+    RUNNING_SIMULATIONS = 1
+
+    def initialize(experiment, resources_interface)
+      @experiment = experiment
+      @resources_interface = resources_interface
+    end
+
+    ##
+    # Returns throughput for given worker sm_uuid. Throughout for worker is calculated as:
+    #   throughput[sim/s] = (finished_simulations + running_simulation)/(Time.now - start_time)
+    def worker_throughput(worker_sm_uuid)
+      calculate_worker_throughput(@resources_interface.get_worker_record_by_sm_uuid(worker_sm_uuid))
+    end
+
+    ##
+    # Returns average throughput for given infrastructure
+    # Arguments:
+    # * infrastructure - hash with infrastructure info
+    # * params:
+    #   * opts
+    #   * cond
+    # Possible cond and opts can be found in MongoActiveRecord#where.
+    # By default only running workers are included into calculations.
+    def infrastructure_throughput(infrastructure, params = {})
+      params[:cond] = Query::RUNNING_WORKERS unless params.has_key? :cond
+      workers_for_infrastructure = @resources_interface.get_workers_records_list(infrastructure, params)
+      workers_for_infrastructure.map {|worker| calculate_worker_throughput(worker)}
+        .reduce(0.0, :+) / Float(workers_for_infrastructure.size)
+    end
+
+    ##
+    # Returns throughput of experiment (system) associated with ExperimentStatistics instance.
+    # System throughput is calculated as sum of all workers throughput [sim/s].
+    # Arguments:
+    # * params:
+    #   * opts
+    #   * cond
+    # Possible cond and opts can be found in MongoActiveRecord#where.
+    # By default only running workers are included into calculations.
+    def system_throughput(params = {})
+      params[:cond] = Query::RUNNING_WORKERS unless params.has_key? :cond
+      @resources_interface.get_enabled_infrastructures.map do |infrastructure|
+        @resources_interface.get_workers_records_list(infrastructure, params)
+          .map {|worker| calculate_worker_throughput worker}
+          .reduce 0.0, :+
+      end .reduce 0.0, :+
```

Comment:
Mało istotne, ale w takich przypadkach to jednak nawias kończący by się przydał. Dla ``reduce`` nie są też używane nawiasy, co już kiedyś omawialiśmy.